### PR TITLE
Implement dual subdivision and weight vectors for tropical variety

### PR DESCRIPTION
**Pull request by verreld7**
This PR introduces some new methods in `TropicalVariety` related to dual subdivision and weight vectors.

### Summary of changes:

1. Add `dual_subdivision` in `TropicalVariety`: return the dual subdivision for any tropical variety as a Graph object
2. Add `_components_of_vertices` in `TropicalCurve`: intermediate method the help calculate weight vectors
3. Add `weight_vectors` in `TropicalCurve`: return the weight vectors of each vertex
4. Add `weight_vectors` in `TropicalSurface`: return the weight vectors of each unique line of intersection
5. Add a few methods related to dual subdivision of tropical curve such as:

    - `is_smooth`
    - `is_simple`
    - `genus`
    - `contribution`

CC: @tscrim

### :memo: Checklist

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [x] I have created tests covering the changes.
- [x] I have updated the documentation and checked the documentation preview.

### :hourglass: Dependencies

#37962: issues related to this PR
#38291: continuation of this PR

### CODE REVIEW

File: src/sage/rings/semirings/tropical_mpolynomial.py [hunk 1]
```diff
@@ -390,6 +405,180 @@ def tropical_variety(self):
             return TropicalSurface(self)
         return TropicalVariety(self)
 
+    def Newton_polytope(self):
```
Comment (tscrim) [inline src/sage/rings/semirings/tropical_mpolynomial.py]:
```suggestion
    def newton_polytope(self):
```
We follow Python's naming conventions, even when the words are proper nouns.

File: src/sage/rings/semirings/tropical_mpolynomial.py [hunk 2]
```diff
@@ -390,6 +405,180 @@ def tropical_variety(self):
...
+            R = PolynomialRing(T, ('x,y'))
+            x, y = R.gen(), R.gen(1)
+            p2 = R(3) + x**2 + R(-2)*y + R(1/2)*x**2*y + R(2)*x*y**3 + R(-1)*x**3*y**4
+            sphinx_plot(p2.dual_subdivision().plot())
+
+        A subdivision with many faces, not all of which are triangles::
+
+            sage: p3 = R(8) + R(4)*x + R(2)*y + R(1)*x^2 + x*y + R(1)*y^2 \
+            ....:     + R(2)*x^3 + x^2*y + x*y^2 + R(4)*y^3 + R(8)*x^4 \
+            ....:     + R(4)*x^3*y + x^2*y^2 + R(2)*x*y^3 + y^4
```
Comment (tscrim) [inline src/sage/rings/semirings/tropical_mpolynomial.py]:
```suggestion
            sage: p3 = (R(8) + R(4)*x + R(2)*y + R(1)*x^2 + x*y + R(1)*y^2
            ....:      + R(2)*x^3 + x^2*y + x*y^2 + R(4)*y^3 + R(8)*x^4
            ....:      + R(4)*x^3*y + x^2*y^2 + R(2)*x*y^3 + y^4)
```
You can see in the built html doc the output is strange. Plus this is considered to be better practice.

File: src/sage/rings/semirings/tropical_mpolynomial.py [hunk 3]
```diff
@@ -390,6 +405,180 @@ def tropical_variety(self):
...
+        .. PLOT::
+            :width: 300 px
+
+            T = TropicalSemiring(QQ,  use_min=False)
+            R = PolynomialRing(T, ('x,y'))
+            x, y = R.gen(), R.gen(1)
+            p3 = R(8) + R(4)*x + R(2)*y + R(1)*x**2 + x*y + R(1)*y**2 \
+                 + R(2)*x**3 + x**2*y + x*y**2 + R(4)*y**3 + R(8)*x**4 \
+                 + R(4)*x**3*y + x**2*y**2 + R(2)*x*y**3 + y**4
+            sphinx_plot(p3.dual_subdivision().plot())
```
Comment (tscrim) [inline src/sage/rings/semirings/tropical_mpolynomial.py]:
```suggestion
            p3 = (R(8) + R(4)*x + R(2)*y + R(1)*x**2 + x*y + R(1)*y**2
                  + R(2)*x**3 + x**2*y + x*y**2 + R(4)*y**3 + R(8)*x**4
                  + R(4)*x**3*y + x**2*y**2 + R(2)*x*y**3 + y**4)
            sphinx_plot(p3.dual_subdivision().plot())
```

File: src/sage/rings/semirings/tropical_polynomial.py [hunk 4]
```diff
@@ -11,10 +11,8 @@
     sage: R.<x> = PolynomialRing(T)
     sage: x.parent()
     Univariate Tropical Polynomial Semiring in x over Rational Field
-    sage: (x + R(3)*x) * (x^2 + x)
-    3*x^3 + 3*x^2
-    sage: (x^2 + R(1)*x + R(-1))^2
-    0*x^4 + 1*x^3 + 2*x^2 + 0*x + (-2)
+    sage: (x^2 + x + R(0))^2
+    0*x^4 + 0*x^3 + 0*x^2 + 0*x + 0
```
Comment (tscrim) [inline src/sage/rings/semirings/tropical_polynomial.py]: Why did you change these tests? Why not just add the extra one?

File: src/sage/rings/semirings/tropical_polynomial.py [hunk 5]
```diff
@@ -168,7 +166,7 @@ def roots(self):
         Return the list of all tropical roots of ``self``, counted with
         multiplicity.
 
-        OUTPUT: a list of tropical numbers
+        OUTPUT: A list of tropical numbers
```
Comment (tscrim) [inline src/sage/rings/semirings/tropical_polynomial.py]: Please revert this as our style is (usually) starting with a lower case in this instance.

File: src/sage/rings/semirings/tropical_variety.py [hunk 6]
```diff
@@ -527,6 +530,211 @@ def update_result(result):
...
+        to the unique intersection lying within that component.
+
+        Assume ``self`` is a `n`-dimensional tropical variety.
+        Suppose `L` is an intersection lying within the components
+        `S_1, ldots, S_k` with respective weights `w_1, ldots, w_k`.
+        This `L` is a linear structure in `\RR^{n-1}` and has `n-1`
+        direction vectors `d_1,d_2,\dots, d_{n-1}`. Each component
+        `S_1, ldots, S_k` has a normal vector `n_1, \ldots, n_k`.
+        Then, we scale each normal vector to an integer vector such
+        that the greatest common divisor of its elements is 1.
```
Comment (tscrim) [inline src/sage/rings/semirings/tropical_variety.py]:
```suggestion
        Assume ``self`` is a `n`-dimensional tropical variety.
        Suppose `L` is an intersection lying within the components
        `S_1, \ldots, S_k` with respective weights `w_1, \ldots, w_k`.
        This `L` is a linear structure in `\RR^{n-1}` and has `n-1`
        direction vectors `d_1,d_2,\ldots, d_{n-1}`. Each component
        `S_1, \ldots, S_k` has a normal vector `n_1, \ldots, n_k`.
        Then, we scale each normal vector to an integer vector such
        that the greatest common divisor of its elements is 1.
```

File: src/sage/rings/semirings/tropical_variety.py [hunk 7]
```diff
@@ -527,6 +530,211 @@ def update_result(result):
...
+                    line_comps[index] = [i]
+                    index += 1
+                else:
+                    match_key = [k for k, v in index_line.items() if v[0] == tuple(new_line)][0]
+                    line_comps[match_key].append(i)
+
+        WV = {i: [] for i in range(len(line_comps))}
+        for k, index in line_comps.items():
+
+            # Calculate direction vector of the line
```
Comment (tscrim) [inline src/sage/rings/semirings/tropical_variety.py]:
```suggestion
        for k, index in line_comps.items():
            # Calculate direction vector of the line
```

File: src/sage/rings/semirings/tropical_variety.py [hunk 8]
```diff
@@ -983,23 +1245,274 @@ def vertices(self):
...
+        of the curve. Each vector corresponds to an edge emanating from
+        that vertex and points in the direction of the edge.
+
+        Suppose `v` is a vertex adjacent to the edges `e_1, ldots, e_k`
+        with respective weights `w_1, ldots, w_k`. Every edge `e_i` is
+        contained in a line (component) defined by an equation. Therefore,
+        there exists a unique integer vector `v_i=(\alpha, \beta)` in
+        the direction of `e_i` such that `\gcd(\alpha, \beta)=1`. Then,
+        each vertex `v` yield the vectors `w_1v_1,ldots,w_kv_k`.
+        These vectors will satisfy the following balancing condition:
```
Comment (tscrim) [inline src/sage/rings/semirings/tropical_variety.py]:
```suggestion
        Suppose `v` is a vertex adjacent to the edges `e_1, \ldots, e_k`
        with respective weights `w_1, \ldots, w_k`. Every edge `e_i` is
        contained in a line (component) defined by an equation. Therefore,
        there exists a unique integer vector `v_i = (\alpha, \beta)` in
        the direction of `e_i` such that `\gcd(\alpha, \beta)=1`. Then,
        each vertex `v` yield the vectors `w_1 v_1, \ldots, w_k v_k`.
        These vectors will satisfy the following balancing condition:
```

File: src/sage/rings/semirings/tropical_variety.py [hunk 9]
```diff
@@ -983,23 +1245,274 @@ def vertices(self):
...
+        return True
+
+    def genus(self):
+        r"""
+        Return the genus of ``self``.
+
+        Let `t(C)` be the number of trivalent vertices, and let `r(C)` be
+        the number of unbounded edges of `C`. The genus of simple tropical
+        curve `C` is defined by the formula:
+        `g(C) = \frac{1}{2}t(C) - \frac{1}{2}r(C) + 1`.
```
Comment (tscrim) [inline src/sage/rings/semirings/tropical_variety.py]:
```suggestion
        curve `C` is defined by the formula:

        .. MATH::

            g(C) = \frac{1}{2}t(C) - \frac{1}{2}r(C) + 1.
```

File: src/sage/rings/semirings/tropical_variety.py [hunk 10]
```diff
@@ -1092,19 +1605,21 @@ def plot(self):
...
             :width: 300 px
 
             T = TropicalSemiring(QQ)
             R = PolynomialRing(T, ('x,y'))
             x, y = R.gen(), R.gen(1)
-            p2 = x**6 + R(4)*x**4*y**2 + R(2)*x**3*y**3 + R(3)*x**2*y**4 + \
-            x*y**5 + R(7)*x**2 + R(5)*x*y + R(3)*y**2 + R(2)*x + y + R(10)
+            p2 = R(8) + R(4)*x + R(2)*y + R(1)*x**2 + x*y + R(1)*y**2 \
+            + R(2)*x**3 + x**2*y + x*y**2 + R(4)*y**3 + R(8)*x**4 \
+            + R(4)*x**3*y + x**2*y**2 + R(2)*x*y**3 + y**4
```
Comment (tscrim) [inline src/sage/rings/semirings/tropical_variety.py]:
```suggestion
            p2 = (R(8) + R(4)*x + R(2)*y + R(1)*x**2 + x*y + R(1)*y**2
             + R(2)*x**3 + x**2*y + x*y**2 + R(4)*y**3 + R(8)*x**4
             + R(4)*x**3*y + x**2*y**2 + R(2)*x*y**3 + y**4)
```

File: src/sage/rings/semirings/tropical_mpolynomial.py [hunk 11]
```diff
@@ -390,6 +405,180 @@ def tropical_variety(self):
...
 
+    def Newton_polytope(self):
+        """
+        Return the Newton polytope of ``self``.
+
+        The Newton polytope is the convex hull of all the points
+        corresponding to the exponents of the monomials of tropical
+        polynomial.
+
+        OUTPUT: :class:`sage.geometry.polyhedron.constructor.Polyhedron`
```
Comment (tscrim) [inline src/sage/rings/semirings/tropical_mpolynomial.py]:
```suggestion
        OUTPUT: :class:`~sage.geometry.polyhedron.constructor.Polyhedron`
```

File: src/sage/rings/semirings/tropical_mpolynomial.py [hunk 12]
```diff
@@ -390,6 +405,180 @@ def tropical_variety(self):
...
+
+        Dual subdivision refers to a specific decomposition of the
+        Newton polytope of a tropical polynomial. The term "dual" is
+        used in the sense that the combinatorial structure of the
+        tropical variety is reflected in the dual subdivision.
+        Specifically, vertices of the dual subdivision correspond to
+        the intersection of multiple components. Edges of the dual
+        subdivision correspond to the individual components.
+
+        OUTPUT: :class:`sage.geometry.polyhedral_complex.PolyhedralComplex`
```
Comment (tscrim) [inline src/sage/rings/semirings/tropical_mpolynomial.py]:
```suggestion
        OUTPUT: :class:`~sage.geometry.polyhedral_complex.PolyhedralComplex`
```

File: src/sage/rings/semirings/tropical_mpolynomial.py [hunk 13]
```diff
@@ -390,6 +405,180 @@ def tropical_variety(self):
...
+        .. PLOT::
+            :width: 300 px
+
+            T = TropicalSemiring(QQ)
+            R = PolynomialRing(T, ('x,y'))
+            x, y = R.gen(), R.gen(1)
+            p1 = x + y
+            sphinx_plot(p1.Newton_polytope().plot())
+
+        Newton polytope in three dimension::
```
Comment (tscrim) [inline src/sage/rings/semirings/tropical_mpolynomial.py]:
```suggestion
        A Newton polytope in three dimension::
```

File: src/sage/rings/semirings/tropical_mpolynomial.py [hunk 14]
```diff
@@ -390,6 +405,180 @@ def tropical_variety(self):
...
+
+        The Newton polytope is the convex hull of all the points
+        corresponding to the exponents of the monomials of tropical
+        polynomial.
+
+        OUTPUT: :class:`sage.geometry.polyhedron.constructor.Polyhedron`
+
+        EXAMPLES:
+
+        Newton polytope for a two-variable tropical polynomial::
```
Comment (tscrim) [inline src/sage/rings/semirings/tropical_mpolynomial.py]:
```suggestion
        A Newton polytope for a two-variable tropical polynomial::
```

File: src/sage/rings/semirings/tropical_variety.py [hunk 15]
```diff
@@ -527,6 +530,211 @@ def update_result(result):
...
+              3: [(0, 1, 1), (0, 0, -1), (0, -1, 0)]})
+
+        Weight vectors of tropical hypersurface::
+
+            sage: T = TropicalSemiring(QQ)
+            sage: R.<a,b,c,d> = PolynomialRing(T)
+            sage: p1 = R(2)*a*b + R(3)*a*c + R(-1)*c^2 + R(-1/3)*a*d
+            sage: tv = p1.tropical_variety()
+            sage: tv.weight_vectors()
+            ({0: ((u1, u3 - 7/3, u3 - 10/3, u3), {u1 <= u3 - 22/3}),
```
Comment (tscrim) [inline src/sage/rings/semirings/tropical_variety.py]: The Conda test is reporting finding a slightly different solution. So it might be better to test the properties (that the balancing condition is satisfied) rather than the explicit output. (Although it is better to demonstrate the balancing condition does hold anyways.)

**Review by tscrim — CHANGES_REQUESTED** [review]
Sorry for taking so long to get to this. Here are my comments.

File: src/sage/rings/semirings/tropical_mpolynomial.py [hunk 16]
```diff
@@ -151,6 +149,23 @@ class TropicalMPolynomial(MPolynomial_polydict):
         p1 = R(3)*a*b + a + R(-1)*b
         sphinx_plot(p1.plot3d())
 
+    Another way to represent tropical curve is through dual subdivision,
+    which is a subdivision of Newton polytope of tropical polynomial::
+
+        sage: p1.Newton_polytope()
```
Comment (tscrim) [inline src/sage/rings/semirings/tropical_mpolynomial.py]:
```suggestion
        sage: p1.newton_polytope()
```

File: src/sage/rings/semirings/tropical_mpolynomial.py [hunk 17]
```diff
@@ -390,6 +405,180 @@ def tropical_variety(self):
...
+        OUTPUT: :class:`~sage.geometry.polyhedron.constructor.Polyhedron`
+
+        EXAMPLES:
+
+        A Newton polytope for a two-variable tropical polynomial::
+
+            sage: T = TropicalSemiring(QQ)
+            sage: R.<x,y> = PolynomialRing(T)
+            sage: p1 = x + y
+            sage: p1.Newton_polytope()
```
Comment (tscrim) [inline src/sage/rings/semirings/tropical_mpolynomial.py]:
```suggestion
            sage: p1.newton_polytope()
```

File: src/sage/rings/semirings/tropical_mpolynomial.py [hunk 18]
```diff
@@ -390,6 +405,180 @@ def tropical_variety(self):
...
+            A 1-dimensional polyhedron in ZZ^2 defined as the convex hull of 2 vertices
+
+        .. PLOT::
+            :width: 300 px
+
+            T = TropicalSemiring(QQ)
+            R = PolynomialRing(T, ('x,y'))
+            x, y = R.gen(), R.gen(1)
+            p1 = x + y
+            sphinx_plot(p1.Newton_polytope().plot())
```
Comment (tscrim) [inline src/sage/rings/semirings/tropical_mpolynomial.py]:
```suggestion
            sphinx_plot(p1.newton_polytope().plot())
```

File: src/sage/rings/semirings/tropical_mpolynomial.py [hunk 19]
```diff
@@ -390,6 +405,180 @@ def tropical_variety(self):
...
+
+        A Newton polytope in three dimension::
+
+            sage: T = TropicalSemiring(QQ)
+            sage: R.<x,y,z> = PolynomialRing(T)
+            sage: p1 = x^2 + x*y*z + x + y + z + R(0)
+            sage: p1.Newton_polytope()
```
Comment (tscrim) [inline src/sage/rings/semirings/tropical_mpolynomial.py]:
```suggestion
            sage: p1.newton_polytope()
```

File: src/sage/rings/semirings/tropical_mpolynomial.py [hunk 20]
```diff
@@ -390,6 +405,180 @@ def tropical_variety(self):
...
+            A 3-dimensional polyhedron in ZZ^3 defined as the convex hull of 5 vertices
+
+        .. PLOT::
+            :width: 300 px
+
+            T = TropicalSemiring(QQ)
+            R = PolynomialRing(T, ('x,y,z'))
+            x, y, z = R.gen(), R.gen(1), R.gen(2)
+            p1 = x**2 + x*y*z + x + y + z + R(0)
+            sphinx_plot(p1.Newton_polytope().plot())
```
Comment (tscrim) [inline src/sage/rings/semirings/tropical_mpolynomial.py]:
```suggestion
            sphinx_plot(p1.newton_polytope().plot())
```

**Review by tscrim — CHANGES_REQUESTED** [review]
I think this is all of them that you did not change the case. Please be sure to run the doctests before pushing it.

File: src/sage/rings/semirings/tropical_polynomial.py [hunk 21]
```diff
@@ -11,10 +11,10 @@
...
     sage: x.parent()
     Univariate Tropical Polynomial Semiring in x over Rational Field
-    sage: (x + R(3)*x) * (x^2 + x)
-    3*x^3 + 3*x^2
-    sage: (x^2 + R(1)*x + R(-1))^2
-    0*x^4 + 1*x^3 + 2*x^2 + 0*x + (-2)
+    sage: (R(3)*x + R(1)) * (x^2 + x)
+    3*x^3 + 3*x^2 + 1*x
+    sage: (x^2 + x + R(0))^2
+    0*x^4 + 0*x^3 + 0*x^2 + 0*x + 0
```
Comment (tscrim) [inline src/sage/rings/semirings/tropical_polynomial.py]: Again, why are you changing these tests? If you want to have different tests, just add them instead.

File: src/sage/rings/semirings/tropical_variety.py [hunk 22]
```diff
@@ -983,29 +1243,283 @@ def vertices(self):
...
+        Return the index of components adjacent to each vertex of ``self``.
+
+        OUTPUT:
+
+        A dictionary where the keys represent the vertices, and the
+        values are lists of tuples. Each tuple consists of the index
+        of an adjacent edge (component) `e_i` and a string indicating
+        the directionality of `e_i` relative to the vertex. The string
+        is either "pos" or "neg", specifying whether it is positive or
+        negative.
```
Comment (tscrim) [inline src/sage/rings/semirings/tropical_variety.py]: Sorry for not realizing this earlier. It probably would be better to return `+1` / `-1` instead of `”pos”` / `”neg”` as it is easier to check.

File: src/sage/rings/semirings/tropical_variety.py [hunk 23]
```diff
@@ -983,29 +1243,283 @@ def vertices(self):
...
+        cov = self._vertices_components()
+        result = {}
+        for vertex in cov:
+            vectors = []
+            for comp in cov[vertex]:
+                weight = self._hypersurface[comp[0]][2]
+                if comp[1] == 'pos':
+                    vectors.append(weight*temp_vectors[comp[0]])
+                else:
+                    vectors.append(weight*(-temp_vectors[comp[0]]))
```
Comment (tscrim) [inline src/sage/rings/semirings/tropical_variety.py]:
```suggestion
                vectors.append(weight*comp[1]*temp_vectors[comp[0]])
```
Plus having it return +/-1 means we can simplify this too.

File: src/sage/rings/semirings/tropical_variety.py [hunk 24]
```diff
@@ -983,29 +1243,283 @@ def vertices(self):
...
+            sage: p1 = x^2 + x + R(1)
+            sage: p1.tropical_variety().is_smooth()
+            False
+            sage: p2 = R(2)*x^2 + x*y + R(2)*y^2 + x + R(-1)*y + R(3)
+            sage: p2.tropical_variety().is_smooth()
+            True
+        """
+        if len(self.vertices()) == self._poly.degree()**2:
+            return True
+        return False
```
Comment (tscrim) [inline src/sage/rings/semirings/tropical_variety.py]:
```suggestion
        return len(self.vertices()) == self._poly.degree() ** 2
```

File: src/sage/rings/semirings/tropical_variety.py [hunk 25]
```diff
@@ -983,29 +1243,283 @@ def vertices(self):
...
+            True
+        """
+        vov = self.weight_vectors()
+        for vertex in self.vertices():
+            if len(vov[vertex]) > 4:
+                return False
+            elif len(vov[vertex]) == 4:
+                for v in vov[vertex]:
+                    if -v not in vov[vertex]:
+                        return False
```
Comment (tscrim) [inline src/sage/rings/semirings/tropical_variety.py]:
```suggestion
            if len(vov[vertex]) == 4:
                if any(-v not in vov[vertex] for v in vov[vertex]):
                    return False
```

File: src/sage/rings/semirings/tropical_variety.py [hunk 26]
```diff
@@ -1092,19 +1606,21 @@ def plot(self):
...
             :width: 300 px
 
             T = TropicalSemiring(QQ)
             R = PolynomialRing(T, ('x,y'))
             x, y = R.gen(), R.gen(1)
-            p2 = x**6 + R(4)*x**4*y**2 + R(2)*x**3*y**3 + R(3)*x**2*y**4 + \
-            x*y**5 + R(7)*x**2 + R(5)*x*y + R(3)*y**2 + R(2)*x + y + R(10)
+            p2 = (R(8) + R(4)*x + R(2)*y + R(1)*x**2 + x*y + R(1)*y**2
+                 + R(2)*x**3 + x**2*y + x*y**2 + R(4)*y**3 + R(8)*x**4
+                 + R(4)*x**3*y + x**2*y**2 + R(2)*x*y**3 + y**4)
```
Comment (tscrim) [inline src/sage/rings/semirings/tropical_variety.py]: Again, rather than changing tests, just add another one. It’s good to have more examples.

**Review by tscrim — CHANGES_REQUESTED** [review]
Thank you. A few other little changes. This should be the last round.

File: src/sage/rings/semirings/tropical_variety.py [hunk 27]
```diff
@@ -527,6 +530,209 @@ def update_result(result):
...
+        unique intersection of the components of tropical variety.
+        Each vector is a normal vector to a component with respect
+        to the unique intersection lying within that component.
+
+        Assume ``self`` is a `n`-dimensional tropical variety.
+        Suppose `L` is an intersection lying within the components
+        `S_1, ldots, S_k` with respective weights `w_1, ldots, w_k`.
+        This `L` is a linear structure in `\RR^{n-1}` and has `n-1`
+        direction vectors `d_1,d_2,\ldots, d_{n-1}`. Each component
+        `S_1, ldots, S_k` has a normal vector `n_1, \ldots, n_k`.
```
Comment (tscrim) [inline src/sage/rings/semirings/tropical_variety.py]:
```suggestion
        `S_1, \ldots, S_k` with respective weights `w_1, \ldots, w_k`.
        This `L` is a linear structure in `\RR^{n-1}` and has `n-1`
        direction vectors `d_1,d_2,\ldots, d_{n-1}`. Each component
        `S_1, \ldots, S_k` has a normal vector `n_1, \ldots, n_k`.
```

File: src/sage/rings/semirings/tropical_variety.py [hunk 28]
```diff
@@ -983,29 +1243,276 @@ def vertices(self):
...
+        Weight vectors are a list of vectors associated with each vertex
+        of the curve. Each vector corresponds to an edge emanating from
+        that vertex and points in the direction of the edge.
+
+        Suppose `v` is a vertex adjacent to the edges `e_1, ldots, e_k`
+        with respective weights `w_1, \ldots, w_k`. Every edge `e_i` is
+        contained in a line (component) defined by an equation. Therefore,
+        there exists a unique integer vector `v_i = (\alpha, \beta)` in
+        the direction of `e_i` such that `\gcd(\alpha, \beta)=1`. Then,
+        each vertex `v` yield the vectors `w_1 v_1, ldots, w_k v_k`.
```
Comment (tscrim) [inline src/sage/rings/semirings/tropical_variety.py]:
```suggestion
        Suppose `v` is a vertex adjacent to the edges `e_1, \ldots, e_k`
        with respective weights `w_1, \ldots, w_k`. Every edge `e_i` is
        contained in a line (component) defined by an equation. Therefore,
        there exists a unique integer vector `v_i = (\alpha, \beta)` in
        the direction of `e_i` such that `\gcd(\alpha, \beta)=1`. Then,
        each vertex `v` yield the vectors `w_1 v_1, \ldots, w_k v_k`.
```

File: src/sage/rings/semirings/tropical_mpolynomial.py [hunk 29]
```diff
@@ -11,8 +11,6 @@
     sage: R.<x,y,z> = PolynomialRing(T)
     sage: z.parent()
     Multivariate Tropical Polynomial Semiring in x, y, z over Rational Field
-    sage: R(2)*x + R(-1)*x + R(5)*y + R(-3)
-    (-1)*x + 5*y + (-3)
```
Comment (tscrim) [inline src/sage/rings/semirings/tropical_mpolynomial.py]: Why did you remove this test?

File: src/sage/rings/semirings/tropical_mpolynomial.py [hunk 30]
```diff
@@ -390,6 +405,180 @@ def tropical_variety(self):
             return TropicalSurface(self)
         return TropicalVariety(self)
 
+    def newton_polytope(self):
+        """
```
Comment (tscrim) [inline src/sage/rings/semirings/tropical_mpolynomial.py]:
```suggestion
        r"""
```

File: src/sage/rings/semirings/tropical_mpolynomial.py [hunk 31]
```diff
@@ -390,6 +405,180 @@ def tropical_variety(self):
...
+
+        EXAMPLES:
+
+        A Newton polytope for a two-variable tropical polynomial::
+
+            sage: T = TropicalSemiring(QQ)
+            sage: R.<x,y> = PolynomialRing(T)
+            sage: p1 = x + y
+            sage: p1.newton_polytope()
+            A 1-dimensional polyhedron in ZZ^2 defined as the convex hull of 2 vertices
```
Comment (tscrim) [inline src/sage/rings/semirings/tropical_mpolynomial.py:428]: You should also have this output the data of the Newton polytope here and in the example below.

File: src/sage/rings/semirings/tropical_mpolynomial.py [hunk 32]
```diff
@@ -390,6 +405,180 @@ def tropical_variety(self):
...
+
+        EXAMPLES:
+
+        Dual subdivision of a tropical curve::
+
+            sage: T = TropicalSemiring(QQ, use_min=False)
+            sage: R.<x,y> = PolynomialRing(T)
+            sage: p1 = R(3) + R(2)*x + R(2)*y + R(3)*x*y + x^2 + y^2
+            sage: p1.dual_subdivision()
+            Polyhedral complex with 4 maximal cells
```
Comment (tscrim) [inline src/sage/rings/semirings/tropical_mpolynomial.py:502]: You should have it also display the maximal cell polytopes in this complex (with their explicit data) and also below.

File: src/sage/rings/semirings/tropical_polynomial.py (hunk 21 above)
Comment (tscrim) [reply src/sage/rings/semirings/tropical_polynomial.py]: You haven't answered why these tests are changed instead of adding additional ones.

**Review by tscrim — CHANGES_REQUESTED** [review]
Also, the bot is reporting that the doc is not building. Have you checked locally?

File: src/sage/rings/semirings/tropical_polynomial.py (hunk 21 above)
Comment (verreld7) [reply src/sage/rings/semirings/tropical_polynomial.py]: The test was changed to make it clearer while testing the same functionality.

**Comment by verreld7** [comment]
The html doc builds properly locally, but the pdf does not. I am still figuring out where the error is.

File: src/sage/rings/semirings/tropical_mpolynomial.py [hunk 33]
```diff
@@ -390,9 +407,250 @@ def tropical_variety(self):
...
 
+    def newton_polytope(self):
+        r"""
+        Return the Newton polytope of ``self``.
+
+        The Newton polytope is the convex hull of all the points
+        corresponding to the exponents of the monomials of tropical
+        polynomial.
+
+        OUTPUT: :class:`~sage.geometry.polyhedron.constructor.Polyhedron`
```
Comment (tscrim) [inline src/sage/rings/semirings/tropical_mpolynomial.py]:
```suggestion
        OUTPUT: :func:`~sage.geometry.polyhedron.constructor.Polyhedron`
```

File: src/sage/rings/semirings/tropical_polynomial.py (hunk 21 above)
Comment (tscrim) [reply src/sage/rings/semirings/tropical_polynomial.py]: I would just have both. More examples/tests is not a bad thing. :)

**Review by tscrim — CHANGES_REQUESTED** [review]
This might fix the pdf docbuild issue; that can be more strict with the type of object linked. Other than that, I don't see any latex or formatting issues.

File: src/sage/rings/semirings/tropical_variety.py [hunk 34]
```diff
@@ -1103,9 +1611,29 @@ def plot(self):
             T = TropicalSemiring(QQ)
             R = PolynomialRing(T, ('x,y'))
             x, y = R.gen(), R.gen(1)
-            p2 = x**6 + R(4)*x**4*y**2 + R(2)*x**3*y**3 + R(3)*x**2*y**4 + \
-            x*y**5 + R(7)*x**2 + R(5)*x*y + R(3)*y**2 + R(2)*x + y + R(10)
+            p2 = (x**6 + R(4)*x**4*y^2 + R(2)*x**3*y**3 + R(3)*x**2*y**4
```
Comment (tscrim) [inline src/sage/rings/semirings/tropical_variety.py]:
```suggestion
            p2 = (x**6 + R(4)*x**4*y**2 + R(2)*x**3*y**3 + R(3)*x**2*y**4
```
Missed `^` to `**`.

**Review by tscrim — APPROVED** [review]
Thank you. That fixed the docbuilding. Positive review.
